core/src/main/scala/Json.scala

```diff
@@ -583,21 +583,14 @@ object Json {
   implicit val HealthCheckEncoder: EncodeJson[health.HealthCheck] =
     EncodeJson[health.HealthCheck] { hs => jString(health.HealthCheck.toString(hs)) }
 
-  implicit lazy val HealthStatusEncoder: EncodeJson[ConsulHealthStatus] =
-    EncodeJson((h: ConsulHealthStatus) =>
-      ("name"    := h.name) ->:
+  implicit lazy val HealthStatusEncoder: EncodeJson[HealthStatus] =
+    EncodeJson((h: HealthStatus) =>
+      ("name"    := h.details.getOrElse("")) ->:
```

Comment:
I'm not sure `""` is the best default. I would either prefer "unspecified" or make this field optional.